content/en/agent/autodiscovery/_index.md

```diff
@@ -30,32 +30,26 @@ Here's a 5-minute video for a bird's eye view of Datadog Agent v6 Autodiscovery
+1. **Create and Load Integration template**: When the Agent starts with Autodiscovery enabled, it loads integration templates from all [available template sources][2]; along with the [Autodiscovery container identifiers][3]. Static configuration files aren't suitable for checks that collect data from ever-changing network endpoints like host or ports, so Autodiscovery uses [**Template Variables**][4] for integration template configuration. Those integration template configurations can be loaded into the Agent in 4 main ways:
 
-### Different Execution
+  * [Using a configuration file mounted within the Agent][5]
+  * [Using Key-Value Store][6]
+  * [Using Kubernetes Annotations][7]
+  * [Using Docker Labels][8]
 
-When the Agent starts with Autodiscovery enabled, it loads check templates from all available template sources&mdash;[not just one or another](#template-source-precedence)&mdash;along with the templates' container identifiers. Unlike in a traditional Agent setup, the Agent doesn't run all checks all the time; it decides which checks to enable by inspecting all containers currently running on the same host as the Agent.
-
-As the Agent inspects each running container, it checks if the container matches any of the container identifiers from any loaded templates. For each match, the Agent generates a static check configuration by substituting the matching container's IP address and port. Then it enables the check using the static configuration.
-
-The Agent watches for Docker events&mdash;container creation, destruction, starts, and stops&mdash;and enables, disables, and regenerates static check configurations on such events.
+2. **Apply an integration template to a specific container**: Unlike in a traditional Agent setup, the Agent doesn't run all checks all the time; it decides which checks to enable by inspecting all containers running on the same host as the Agent and the corresponding loaded integration templates. The Agent then watches for Docker events&mdash;container creation, destruction, starts, and stops&mdash;and enables, disables, and regenerates static check configurations on such events. As the Agent inspects each running container, it checks if the container matches any of the [Autodiscovery container identifiers][3] from any loaded integration templates. For each match, the Agent generates a static check configuration by substituting the [Template Variables][9] with the matching container's specific values. Then it enables the check using the static configuration.
 
 ## How to set it up
 
-The Datadog Docker Agent automatically auto-discovers other containers.
+**The Datadog Docker Agent automatically auto-discovers other containers**.
 
-To use Autodiscovery on the Datadog **Host** Agent, add these settings to your `datadog.yaml` configuration file:
+To use Autodiscovery on the Datadog **Host** Agent, add these settings to your [`datadog.yaml` configuration file][10]:
```

Comment:
the snippets bellow works for *docker* but we have different configurations depending on the environment:
* https://github.com/DataDog/datadog-agent/blob/master/Dockerfiles/agent/datadog-docker.yaml
* https://github.com/DataDog/datadog-agent/blob/master/Dockerfiles/agent/datadog-ecs.yaml
* https://github.com/DataDog/datadog-agent/blob/master/Dockerfiles/agent/datadog-kubernetes.yaml
* https://github.com/DataDog/datadog-agent/blob/master/Dockerfiles/agent/datadog-k8s-docker.yaml